data/binding/bindlists.go

```diff
@@ -0,0 +1,239 @@
+}
+
+func (l *boundBoolList) Get(i int) bool {
+	if i > l.Length() {
+		return false
+	}
+
+	return l.GetItem(i).(Bool).Get()
+}
+
+func (l *boundBoolList) Prepend(val bool) {
+	l.prependItem(BindBool(&val))
+}
+
+func (l *boundBoolList) Set(i int, v bool) {
+	if i > l.Length() {
+		return
+	}
+
+	l.GetItem(i).(Bool).Set(v)
+}
+
+// FloatList supports binding a list of float64 values.
+//
+// Since: 2.0.0
+type FloatList interface {
+	DataList
+
+	Append(float64)
+	Get(int) float64
+	Prepend(float64)
+	Set(int, float64)
+}
+
+// NewFloatList returns a bindable list of float64 values.
+//
+// Since: 2.0.0
+func NewFloatList() FloatList {
+	return &boundFloatList{}
+}
+
+type boundFloatList struct {
+	listBase
+}
+
+func (l *boundFloatList) Append(val float64) {
+	l.appendItem(BindFloat(&val))
+}
+
+func (l *boundFloatList) Get(i int) float64 {
+	if i > l.Length() {
+		return 0.0
+	}
+
+	return l.GetItem(i).(Float).Get()
+}
+
+func (l *boundFloatList) Prepend(val float64) {
+	l.prependItem(BindFloat(&val))
+}
+
+func (l *boundFloatList) Set(i int, v float64) {
+	if i > l.Length() {
+		return
+	}
+
+	l.GetItem(i).(Float).Set(v)
+}
+
+// IntList supports binding a list of int values.
+//
+// Since: 2.0.0
+type IntList interface {
+	DataList
+
+	Append(int)
+	Get(int) int
+	Prepend(int)
+	Set(int, int)
+}
+
+// NewIntList returns a bindable list of int values.
+//
+// Since: 2.0.0
+func NewIntList() IntList {
+	return &boundIntList{}
+}
+
+type boundIntList struct {
+	listBase
+}
+
+func (l *boundIntList) Append(val int) {
+	l.appendItem(BindInt(&val))
+}
+
+func (l *boundIntList) Get(i int) int {
+	if i > l.Length() {
+		return 0
+	}
+
+	return l.GetItem(i).(Int).Get()
+}
+
+func (l *boundIntList) Prepend(val int) {
+	l.prependItem(BindInt(&val))
+}
+
+func (l *boundIntList) Set(i int, v int) {
+	if i > l.Length() {
+		return
+	}
+
+	l.GetItem(i).(Int).Set(v)
+}
+
+// RuneList supports binding a list of rune values.
+//
+// Since: 2.0.0
+type RuneList interface {
+	DataList
+
+	Append(rune)
+	Get(int) rune
+	Prepend(rune)
+	Set(int, rune)
+}
+
+// NewRuneList returns a bindable list of rune values.
+//
+// Since: 2.0.0
+func NewRuneList() RuneList {
+	return &boundRuneList{}
+}
+
+type boundRuneList struct {
+	listBase
+}
+
+func (l *boundRuneList) Append(val rune) {
+	l.appendItem(BindRune(&val))
+}
+
+func (l *boundRuneList) Get(i int) rune {
+	if i > l.Length() {
+		return rune(0)
+	}
+
+	return l.GetItem(i).(Rune).Get()
+}
+
+func (l *boundRuneList) Prepend(val rune) {
+	l.prependItem(BindRune(&val))
+}
+
+func (l *boundRuneList) Set(i int, v rune) {
+	if i > l.Length() {
+		return
+	}
+
+	l.GetItem(i).(Rune).Set(v)
+}
+
+// StringList supports binding a list of string values.
+//
+// Since: 2.0.0
+type StringList interface {
+	DataList
+
+	Append(string)
+	Get(int) string
+	Prepend(string)
+	Set(int, string)
+}
+
+// NewStringList returns a bindable list of string values.
+//
+// Since: 2.0.0
+func NewStringList() StringList {
```

Comment:
We can't bind to an existing slice
```go
strings := []string{"a", "b", "c"}
bind := NewStringList(strings)
```
As this is something we want to support longer term, let's articulate the major hurdles; I would guess the StringList cannot determine if the slice size (1), or ordering (2) has changed, or if individual elements have changed (3);
```go
strings = append(strings, "d") // (1)
strings[0], strings[2] = strings[2], strings[1] // (2)
strings[0]+="aa" // (3)
``` 
Are there any I'm missing?

Reply:
Opened and added other thoughts alongside this to #1645